modules/extensions/src/types/index.ts

```diff
@@ -188,8 +191,13 @@ export interface ExperimentalAPI {
       error: string | null;
     }>;
   }>;
+  editor: Comlink.RemoteObject<{
+    getPreferences: () => Promise<EditorPreferences>;
+  }>;
 }
 
 export type ExtensionPort = Comlink.Remote<ExtensionPortAPI> & {
-  experimental: Comlink.RemoteObject<ExperimentalAPI>;
+  experimental: Comlink.RemoteObject<ExperimentalAPI> & {
+    editor: Comlink.RemoteObject<ExperimentalAPI["editor"]>;
+  };
```

Comment:
if you rebase on main, you'll pick up the new types. that should make all this a lot easier